ui/src/components/DeleteModal/DeleteModal.test.tsx

```diff
@@ -0,0 +1,71 @@
+import { render, screen } from '@testing-library/react';
+import React, { createContext } from 'react';
+import userEvent from '@testing-library/user-event';
+import { AxiosResponse } from 'axios';
+import * as axiosWrapper from '../../util/axiosCallWrapper';
+import DeleteModal from './DeleteModal';
+import { setUnifiedConfig } from '../../util/util';
+import { UnifiedConfig } from './DeleteModal.stories';
+
+jest.mock('immutability-helper', () => {
+    return () => 1;
+});
+
+describe('Text Area Component', () => {
+    const handleClose = jest.fn();
+
+    const TableContext = createContext({
+        rowData: { serviceName: { stanzaName: 1 } },
+        setRowData: () => {},
+    });
+
+    beforeEach(() => {
+        setUnifiedConfig(UnifiedConfig);
+
+        render(
+            <TableContext.Provider
+                value={{ rowData: { serviceName: { stanzaName: 1 } }, setRowData: () => {} }}
+            >
+                <DeleteModal
+                    handleRequestClose={handleClose}
+                    serviceName="serviceName"
+                    stanzaName="stanzaName"
+                    page="inputs"
+                    open
+                />
+            </TableContext.Provider>
+        );
+    });
+
+    afterAll(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should render delete modal correctly', () => {
+        const deleteModal = screen.getByTestId('modal');
+        expect(deleteModal).toBeInTheDocument();
+
+        const buttons = screen.getAllByTestId('button');
+
+        expect(buttons[0]).toHaveTextContent('Cancel');
```

Comment:
Too fragile. what if button order is changed? 
```suggestion
        expect(screen.getByRole('button', { name: /cancel/i })).toBeInTheDocument();
```

Reply:
https://testing-library.com/docs/queries/about#priority

Reply:
replaced it with 
```
        const cancelButton = screen.getByRole('button', { name: /cancel/i });
        expect(cancelButton).toBeInTheDocument();

        const deleteButton = screen.getByRole('button', { name: /delete/i });
        expect(deleteButton).toBeInTheDocument();
```